armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -265,8 +265,20 @@ def construct(
         b.p.xsType = xsType
         b.setBuLimitInfo()
         b = self._mergeComponents(b)
+
+        if spatialGrid:
+            b.spatialGrid = spatialGrid
+        else:
+            try:
+                cornersUpCore = (
+                    blueprint.gridDesigns[blueprint.systemDesigns["core"].gridName].geom
+                    == geometry.HEX_CORNERS_UP
+                )
+                b.autoCreateSpatialGrids(cornersUp=not cornersUpCore)
+            except (TypeError, ValueError, NotImplementedError) as e:
+                runLog.extra(str(e), single=True)
+
```

Comment:
Instead of using `orientBlocks()`, this is where we now will add the spatial grid to individual blocks. This ensures that all blocks (to the extent possible) have spatial grids which are aligned with the corners-up vs flats-up state of the core.